service/history/api/workflow_id_dedup.go

```diff
@@ -100,6 +104,39 @@ func ResolveDuplicateWorkflowID(
 	return nil, nil
 }
 
+func resolveJustStartedDuplicatedWorkflows(
```

Comment:
I'm not sure `resolveJustStarted` expresses this clearly. How about `throttledTerminateWorkflowAction`?

Reply:
I don't think throttle express it either. We basically make a choice which one to "let live".
I agree, naming is not ideal...
I will rename this to 'resolveDuplicateWorkflowStart'

Reply:
> We basically make a choice which one to "let live".

Isn't that the definition of throttling? :D

Reply:
May be you are right, but it somehow doesn't fit into my mental model. For me throttling is usually something to do with rate of change. And in this case decision is based on time...
I will leave it as is.